spec/aws-record/record_spec.rb

```diff
@@ -164,5 +164,72 @@ module Aws
       end
     end
 
+    describe 'inheritance support for table name' do
+      let(:parent_model) do
+        Class.new do
+          include(Aws::Record)
+          set_table_name('ParentTable')
+        end
+      end
+
+      let(:child_model) do
+        Class.new(parent_model) do
+          include(Aws::Record)
+        end
+      end
+
+      it 'should have child model inherit table name from parent model if it is defined in parent model'  do
+        expect(parent_model.table_name).to eq('ParentTable')
+        expect(child_model.table_name).to eq('ParentTable')
+      end
+
+      it 'should have child model override parent table name if defined in model' do
+        child_model.set_table_name('ChildTable')
+        expect(parent_model.table_name).to eq('ParentTable')
+        expect(child_model.table_name).to eq('ChildTable')
+      end
+
+      it 'should have parent and child models maintain their own table names' do
```

Comment:
rather than saying "own table names" - we can be explicit that this is for default table names (ie, I think changing "own" to "default" will make this a bit more clear).

Reply:
Complete!